six/modules/c++/cphd/include/cphd/FileHeader.h

```diff
@@ -28,171 +27,203 @@
+     *
+     *  \return std::string Returns header as string
+     */
+    std::string toString() const override;
+
+    /*
+     *  \func set
+     *
+     *  \brief Set the file header
+     *
+     *  (Wrapper for actual set function) Set the file header
+     *  to its final state compute the offsets of the sections that
+     *  depend on the file header size
+     *
+     *  \return size_t Return size of header
+     */
     size_t set();
-    size_t set(sys::Off_T xmlSize, sys::Off_T vbmSize, sys::Off_T cphdSize);
 
-    // Required elements
-    void setXMLsize(sys::Off_T size)
-    {
-        mXmlDataSize = size;
-    }
+    /*
+     *  \func set
+     *
+     *  \brief Set the file header to its final state
+     *
+     *  Set the file header to its final state
+     *  compute the offsets of the sections that depend
+     *  on the file header size
+     *  Adds padding to modulo 8 for pvp block
+     *
+     *  \param xmlBlockSize Size of XML block
+     *  \param supportBlockSize Size of support block
+     *  \param pvpBlockSize Size of pvp block
+     *  \param signalBlockSize Size of signal block
+     *
+     *  \return size_t Return size of header
+     */
+    size_t set(sys::Off_T xmlBlockSize, sys::Off_T supportBlockSize,
+            sys::Off_T pvpBlockSize, sys::Off_T signalBlockSize);
 
-    void setVBMsize(sys::Off_T size)
+    //! Set required elements
+    void setXMLBlockSize(sys::Off_T size)
     {
-        mVbDataSize = size;
+        mXmlBlockSize = size;
     }
-
-    void setCPHDsize(sys::Off_T size)
+    void setPvpBlockSize(sys::Off_T size)
     {
-        mCphdDataSize = size;
+        mPvpBlockSize = size;
     }
-
-    void setXMLoffset(sys::Off_T offset)
+    void setSignalBlockSize(sys::Off_T size)
     {
-        mXmlByteOffset = offset;
+        mSignalBlockSize = size;
     }
-
-    void setVBMoffset(sys::Off_T offset)
+    void setXMLBlockByteOffset(sys::Off_T offset)
     {
-        mVbByteOffset = offset;
+        mXmlBlockByteOffset = offset;
     }
-
-    void setCPHDoffset(sys::Off_T offset)
+    void setPvpBlockByteOffset(sys::Off_T offset)
     {
-        mCphdByteOffset = offset;
+        mPvpBlockByteOffset = offset;
     }
-
-    sys::Off_T getXMLsize() const
+    void setSignalBlockByteOffset(sys::Off_T offset)
     {
-        return mXmlDataSize;
+        mSignalBlockByteOffset = offset;
     }
-
-    sys::Off_T getVBMsize() const
+    void setClassification(const std::string& str)
     {
-        return mVbDataSize;
+        mClassification = str;
     }
-
-    sys::Off_T getCPHDsize() const
+    void setReleaseInfo(const std::string& str)
     {
-        return mCphdDataSize;
+        mReleaseInfo = str;
     }
 
-    sys::Off_T getXMLoffset() const
+    //! Get required elements
+    sys::Off_T getXMLBlockSize() const
     {
-        return mXmlByteOffset;
+        return mXmlBlockSize;
     }
-
-    sys::Off_T getVBMoffset() const
+    sys::Off_T getPvpBlockSize() const
```

Comment:
Do I need to comment each getter/setter function?

Reply:
"Trivial get/set methods need not be commented as long as the property they control is."